js/main.js

```diff
@@ -62,140 +62,143 @@ function agentDetails(event) {
 
           var $abilityThreeName = document.createElement('p');
           $abilityThreeName.setAttribute('class', 'ability-name ability-three-name noto-font');
-          $abilityThreeName.textContent = xhr.response.data[i].abilities[2].displayName;
           $abilityThreeContainer.appendChild($abilityThreeName);
 
           var $abilityThreeText = document.createElement('p');
           $abilityThreeText.setAttribute('class', 'ability-description ability-description-three');
-          $abilityThreeText.textContent = xhr.response.data[i].abilities[2].description;
           $abilityThreeContainer.appendChild($abilityThreeText);
 
           var $abilityFourIcon = document.createElement('img');
           $abilityFourIcon.setAttribute('class', 'ability-four-icon ability');
-          $abilityFourIcon.setAttribute('src', xhr.response.data[i].abilities[3].displayIcon);
           $abilityFourIconContainer.appendChild($abilityFourIcon);
 
           var $abilityFourName = document.createElement('p');
           $abilityFourName.setAttribute('class', 'ability-name ability-four-name noto-font');
-          $abilityFourName.textContent = xhr.response.data[i].abilities[3].displayName;
           $abilityFourContainer.appendChild($abilityFourName);
 
           var $abilityFourText = document.createElement('p');
           $abilityFourText.setAttribute('class', 'ability-description ability-description-four');
-          $abilityFourText.textContent = xhr.response.data[i].abilities[3].description;
           $abilityFourContainer.appendChild($abilityFourText);
 
+          for (var j = 0; j < xhr.response.data[id].abilities.length; j++) {
+            if (j === 0) {
+              $abilityOneIcon.setAttribute('src', xhr.response.data[id].abilities[j].displayIcon);
+              $abilityOneName.textContent = xhr.response.data[id].abilities[j].displayName;
+              $abilityOneText.textContent = xhr.response.data[id].abilities[j].description;
+            } else if (j === 1) {
+              $abilityTwoIcon.setAttribute('src', xhr.response.data[id].abilities[j].displayIcon);
+              $abilityTwoName.textContent = xhr.response.data[id].abilities[j].displayName;
+              $abilityTwoText.textContent = xhr.response.data[id].abilities[j].description;
+            } else if (j === 2) {
+              $abilityThreeIcon.setAttribute('src', xhr.response.data[id].abilities[j].displayIcon);
+              $abilityThreeName.textContent = xhr.response.data[id].abilities[j].displayName;
+              $abilityThreeText.textContent = xhr.response.data[id].abilities[j].description;
+            } else if (j === 3) {
+              $abilityFourIcon.setAttribute('src', xhr.response.data[id].abilities[j].displayIcon);
+              $abilityFourName.textContent = xhr.response.data[id].abilities[j].displayName;
+              $abilityFourText.textContent = xhr.response.data[id].abilities[j].description;
+            }
```

Comment:
This is still the same problem. You should be able to make the elements, configure the elements, and append the elements inside of this loop. There should be no conditional necessary.